packages/http-client-java/emitter/src/code-model-builder.ts

```diff
@@ -489,106 +489,120 @@ export class CodeModelBuilder {
             }
-          } else if (initializationProperty.type.kind === "endpoint") {
-            sdkPathParameters = initializationProperty.type.templateArguments;
-            baseUri = initializationProperty.type.serverUrl;
+          } else if (initializationProperty.type.variantTypes.length > 2) {
+            throw new Error("Multiple server url defined for one client is not supported yet.");
           }
-
-          hostParameters = this.processHostParameters(sdkPathParameters);
-          codeModelClient.addGlobalParameters(hostParameters);
+        } else if (initializationProperty.type.kind === "endpoint") {
+          sdkPathParameters = initializationProperty.type.templateArguments;
+          baseUri = initializationProperty.type.serverUrl;
         }
-      });
-
-      const clientContext = new ClientContext(
-        baseUri,
-        hostParameters,
-        codeModelClient.globalParameters!,
-        codeModelClient.apiVersions,
-      );
 
-      // preprocess operation groups and operations
-      // operations without operation group
-      const serviceMethodsWithoutSubClient = this.listServiceMethodsUnderClient(client);
-      let codeModelGroup = new OperationGroup("");
-      for (const serviceMethod of serviceMethodsWithoutSubClient) {
-        if (!this.needToSkipProcessingOperation(serviceMethod.__raw, clientContext)) {
-          codeModelGroup.addOperation(this.processOperation(serviceMethod, clientContext, ""));
-        }
+        hostParameters = this.processHostParameters(sdkPathParameters);
+        codeModelClient.addGlobalParameters(hostParameters);
       }
-      if (codeModelGroup.operations?.length > 0) {
-        codeModelClient.operationGroups.push(codeModelGroup);
+    });
+
+    const clientContext = new ClientContext(
+      baseUri,
+      hostParameters,
+      codeModelClient.globalParameters!,
+      codeModelClient.apiVersions,
+    );
+
+    const enableSubclient: boolean = Boolean(this.options["enable-subclient"]);
+
+    // preprocess operation groups and operations
+    // operations without operation group
+    const serviceMethodsWithoutSubClient = this.listServiceMethodsUnderClient(client);
+    let codeModelGroup = new OperationGroup("");
+    for (const serviceMethod of serviceMethodsWithoutSubClient) {
+      if (!this.needToSkipProcessingOperation(serviceMethod.__raw, clientContext)) {
+        codeModelGroup.addOperation(this.processOperation(serviceMethod, clientContext, ""));
       }
+    }
+    if (codeModelGroup.operations?.length > 0 || enableSubclient) {
+      codeModelClient.operationGroups.push(codeModelGroup);
+    }
 
+    const subClients = this.listSubClientsUnderClient(client, !enableSubclient);
+    if (enableSubclient) {
+      // subclient, no operation group
+      for (const subClient of subClients) {
+        const codeModelSubclient = this.processClient(subClient);
+        codeModelClient.addSubClient(codeModelSubclient);
+      }
+    } else {
```

Comment:
Most is refactor (extract code to `processClient`, as we now need to call it recursively).

Here is the main change, with the option, we would not do operation group, and every client is a client (or subclient).